### PR TITLE
Fix to HowTo examples

### PR DESCRIPTION
**Pull request by RichardWallis**
Implements fix for issue (#1884)

### CODE REVIEW

**Comment by danbri** [comment]
Thanks, though not gonna rush this one. There are unconventional currencies other than "crypto" currencies, https://en.wikipedia.org/wiki/Local_exchange_trading_system#Examples that we maybe ought to handle in common manner.

**Comment by danbri** [comment]
Looking at this with a colleague, we are concerned the steps section is broken, but will look at a followup edit after merging as the other fixes are good.

File: data/sdo-howto-examples.txt [hunk 1]
```diff
@@ -92,98 +92,342 @@ PRE-MARKUP:
...
-        <div itemprop="text">Put the equipment and the flat tire away.</div>
-      </div>
+    <strong><span itemprop="name">Change a Flat Tire</span></strong>
+    <div>About <span itemprop="estimatedCost" itemscope itemtype="http://schema.org/MonetaryAmount">
+        <meta itemprop="currency" content="USD"></meta>
+        <meta itemprop="value" content="20"></meta>
+    </span>$20</div>
+    <div>About <span itemprop="totalTime" content="PT30M">30 minutes</span></div>
+    <div>Necessary Items:</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Spare tire</div>
```
Comment (dbiollo) [inline data/sdo-howto-examples.txt]: I believe the name of the HowToTool will not be set with this syntax. The previous way works, or if you remove the itemscope & itemtype.

**Comment by RichardWallis** [comment]
@dbiollo Good point.

There are two approaches to getting the name of the HowToTool, and HowToSuppy, to be picked out.
```
<div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Spare tire</div>
```
or
```
<div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">
     <span itemprop="name">Spare tire</span>
</div>
```
My preference is for the second of these as it could be expanded thus:
```
<div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">
     <span itemprop="name">Spare tire</span>
     <img itemprop="image" src="spare-tire.jpg" />
</div>
```
I will raise a new issue to explore this fix and the concerns raised by @danbri

File: data/sdo-howto-examples.txt [hunk 2]
```diff
@@ -92,98 +92,342 @@ PRE-MARKUP:
...
-        <div itemprop="text">Replace the hubcap.</div>
-      </div>
-    </div>
-    <div itemprop="itemListElement" itemscope itemtype="http://schema.org/HowToStep">
-      <div itemprop="itemListElement" itemscope itemtype="http://schema.org/HowToDirection">
-        <div itemprop="text">Put the equipment and the flat tire away.</div>
-      </div>
+    <strong><span itemprop="name">Change a Flat Tire</span></strong>
+    <div>About <span itemprop="estimatedCost" itemscope itemtype="http://schema.org/MonetaryAmount">
+        <meta itemprop="currency" content="USD"></meta>
```
Comment (dbiollo) [inline data/sdo-howto-examples.txt]: There is </meta> close tag, meta is an empty element.

File: data/sdo-howto-examples.txt [hunk 3]
```diff
@@ -92,98 +92,342 @@ PRE-MARKUP:
...
+        <meta itemprop="value" content="20"></meta>
+    </span>$20</div>
+    <div>About <span itemprop="totalTime" content="PT30M">30 minutes</span></div>
+    <div>Necessary Items:</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Spare tire</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Lug wrench</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Jack</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Wheel wedges</div>
+    <div itemprop="supply" itemscope itemtype="http://schema.org/HowToSupply">Flares</div>
+    <div itemprop="steps" itemscope itemtype="http://schema.org/ItemList">
```
Comment (dbiollo) [inline data/sdo-howto-examples.txt]: The field name is now called "step"

File: data/sdo-howto-examples.txt [hunk 4]
```diff
@@ -92,98 +92,342 @@ PRE-MARKUP:
...
+                <div itemprop="itemListElement" itemscope itemtype="http://schema.org/HowToDirection">
+                    <meta itemprop="position" content="1"></meta>
+                    <img itemprop="duringMedia" href="lit-flares.jpg" />
+                    <div itemprop="text">Turn on your hazard lights and set the flares.</div>
+                </div>
+                <div itemprop="itemListElement" itemscope itemtype="http://schema.org/HowToTip">
+                    <meta itemprop="position" content="2"></meta>
+                    <div itemprop="text">You're going to need space and want to be visible.</div>
+                </div>
+                <div itemprop="itemListElement" itemscope itemtype="http://schema.org/HowToDirection">
```
Comment (dbiollo) [inline data/sdo-howto-examples.txt]: I believe this changes the meaning of the HowTo Steps. Originally there was a HowToStep wrapping each HowToDirection. My understanding is that a HowToStep represents a (usually numbered) significant step, which can contain the HowToDirection primary text and optionally a HowToTip (a side note for more complex steps). 

Was there any intent / discussion around changing the design of the HowTo for this release? If not, I'd suggest keeping the original HowToStep structure (wrapping the direction + tip) for now.

File: data/sdo-howto-examples.txt (hunk 3 above)
Comment (dbiollo) [reply data/sdo-howto-examples.txt]: Also "step" can be CreativeWork, HowToSection, HowToStep, Text. So I don't think ItemList is valid is it?

If the ItemList wrapper is removed then the sections can be steps like so:

```<div itemprop="step" itemscope itemtype="http://schema.org/HowToSection">```

File: data/sdo-howto-examples.txt [hunk 5]
```diff
@@ -92,98 +92,342 @@ PRE-MARKUP:
...
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Spare tire</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Lug wrench</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Jack</div>
+    <div itemprop="tool" itemscope itemtype="http://schema.org/HowToTool">Wheel wedges</div>
+    <div itemprop="supply" itemscope itemtype="http://schema.org/HowToSupply">Flares</div>
+    <div itemprop="steps" itemscope itemtype="http://schema.org/ItemList">
+        <div itemprop="itemListElement" itemscope itemtype="http://schema.org/HowToSection">
+            <div itemprop="name">Preparation</div>
+            <meta itemprop="position" content="1"></meta>
+            <div itemprop="steps" itemscope itemtype="http://schema.org/HowToStep">
```
Comment (dbiollo) [inline data/sdo-howto-examples.txt]: HowToSection does not have a "steps" property, but it is an ItemList so this itemprop should be "itemListElement".

**Comment by dbiollo** [comment]
Took a closer look and left some comments on things that look problematic.

**Comment by RichardWallis** [comment]
Changes made and now in PR (#1902)

**Pull request merged**